src/comms.rs

```diff
@@ -620,17 +580,117 @@ pub enum UnitStatus {
+}
+
+impl fmt::Display for UnitHealth {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         f.write_str(match *self {
-            UnitStatus::Healthy => "healthy",
-            UnitStatus::Stalled => "stalled",
-            UnitStatus::Gone => "gone",
+            UnitHealth::Healthy => "healthy",
+            UnitHealth::Stalled => "stalled",
+            UnitHealth::Gone => "gone",
         })
     }
 }
 
 
+//------------ UnitStatus ----------------------------------------------------
+
+/// A combination of both a unit’s health and the last payload update.
+///
+/// This is a helper type that makes it easier to apply updates.
+#[derive(Clone, Debug, Default)]
+struct UnitStatus {
+    /// The current health of the unit.
+    health: UnitHealth,
+
+    /// The last payload update if there ever was one.
+    payload: Option<payload::Update>,
+}
+
+impl UnitStatus {
+    /// Applies a unit update to the status.
+    ///
+    /// Returns whether the status has changed.
+    fn apply(&mut self, update: &UnitUpdate) -> bool {
+        match update {
+            UnitUpdate::Payload(payload) => {
+                if matches!(self.health, UnitHealth::Healthy)
+                    && Some(payload) == self.payload.as_ref()
+                {
+                    false
+                }
+                else {
+                    self.health = UnitHealth::Healthy;
+                    self.payload = Some(payload.clone());
+                    true
+                }
+            }
+            UnitUpdate::Stalled => {
+                if matches!(self.health, UnitHealth::Stalled) {
+                    false
+                }
+                else {
+                    self.health = UnitHealth::Stalled;
+                    true
+                }
+            }
+            UnitUpdate::Gone => {
+                if matches!(self.health, UnitHealth::Gone) {
+                    false
+                }
+                else {
+                    self.health = UnitHealth::Gone;
+                    true
+                }
+            }
+        }
+    }
+
+    /// Returns an update corresponding with the current unit status.
+    ///
+    /// This may be `None` if the unit status indicates that there hasn’t
+    /// been an update yet.
+    fn to_update(&self) -> Option<UnitUpdate> {
+        match self.health {
+            UnitHealth::Healthy => {
+                self.payload.as_ref().map(|payload| {
+                    UnitUpdate::Payload(payload.clone())
+                })
+            }
+            UnitHealth::Stalled => Some(UnitUpdate::Stalled),
+            UnitHealth::Gone => Some(UnitUpdate::Gone),
+        }
+    }
+}
+
+
+//------------ UnitUpdate ----------------------------------------------------
+
+/// An update to the unit.
+#[derive(Clone, Debug)]
+pub enum UnitUpdate {
+    /// A new payload set has become available.
+    ///
+    /// This also implies a change to unit status “healthy” if that is not the
```

Comment:
You could simplify this comment by saying "This implies that the unit is healthy".

Reply:
I wanted to express that this equals a change in health if necessary.